### DISCUSSION TITLE
Configuration files affecting database connection speed?

### DISCUSSION THREAD

**Discussion by sebastian-de** [pgmodeler/pgmodeler#1920] [Q&A] [answered]
Hi,

I've used pgModeler for a few weeks now and I think it's awesome, thanks a lot for creating it!

Now I have a very strange problem and am hoping that someone can help me. Because the problem is so weird, this is going to be a longer post to explain it properly - sorry for that.

A few days ago, all connections from pgModeler to PostgreSQL became really slow. For example, when I select a new table in the data manipulation grid, it takes up to 4-6 seconds until the table loads, even if it's really small or empty:

![screenshot](https://github.com/user-attachments/assets/3c2c42eb-3a83-4b87-b595-369591458baf)

This only happens when connecting to Postgres from pgModeler, other clients work fine (I ran pgbench - all good).

I'm using Fedora 41 with Plasma 6 as DE. Postgres is v16.4 in a podman container. pgModeler in Fedora currently is v1.1.3.
I didn't use pgModeler for a few days before I noticed this, so I can't pin down exactly, what changed on my system during that time (I install updates regularly).

To arrow down the cause, I
- recreated the Postgres container (and the data directory)
- tried different Postgres versions
- used the Fedora Postgres packages, instead of running it in podman
- compiled and ran pgModeler v1.1.5
- removed `~/.config/pgmodeler-1.0` and `~/.local/state/pgmodelerstaterc` before running it again

Nothing eliminated the problem - the connection from pgModeler to Postgres remained slow.

The last thing I did: I created a test user with a clean home directory and ran the same setup there (Postgres 16 in podman, pgModeler v1.1.3).
And indeed: There the database connection was as fast as it was in the beginning!

So the cause for the slow connections must be somewhere in the configuration of my normal user.

What other configurations, other than `~/.config/pgmodeler-1.0` might be relevant?
Or is there a way to debug/trace the database connection in pgModeler in some way?

I'm thankful for any ideas to narrow this further down!

**Comment by rkhaotix** [comment]
Hi @sebastian-de 

That's really strange. Never faced something like this. 

Does connecting to another database server (not in your machine ) has the same undesired effect?

Are you browsing tables that has long text (thousand of bytes) in their columns ? 

The only configuration directory is the one you mentioned, and I doubt that the configuration files are the culprits.

**Reply by sebastian-de** [reply]
Thanks for your reply!

> Does connecting to another database server (not in your machine ) has the same undesired effect?

That's a good idea, I'll try that later.

> Are you browsing tables that has long text (thousand of bytes) in their columns ?

No, the table I loaded for the screenshot only has a single `varchar(10)` column.

> The only configuration directory is the one you mentioned, and I doubt that the configuration files are the culprits.

A suspicion I have is that maybe Plasma interferes in some way - I can imagine that it alters the Qt configuration in some way. But that will be hard to pin down, too.

**Reply by sebastian-de** [reply]
> Does connecting to another database server (not in your machine ) has the same undesired effect?

Yes, connecting to a remote database server is slow as well.

**Comment by sebastian-de** [comment, marked as answer]
OK, I found the cause: It was Kerberos.

I ran pgModeler trough strace and saw a lot of connections to `/var/run/.heim_org.h5l.kcm-socket` even though I don't have SSL or GSSAPI enabled for my test database.

And now I remember what I did, right before the connections became slow: I installed Fedora packaging tools and [acquired a Fedora Kerberos ticket](https://docs.fedoraproject.org/en-US/package-maintainers/Installing_Packager_Tools/#kerberos_ticket), so from that point on sssd-kcm had credentials stored for my user. My test user doesn't have any credentials stored, hence it wasn't affected.

I get a fast connection, if I add `gssencmode=disable ` to "Other params" (I had to explicitly add the whitespace at the end).

Phew, now I can sleep again - this was driving me nuts ;)
